src/Features/LanguageServer/Protocol/Handler/Definitions/AbstractGoToDefinitionHandler.cs

```diff
@@ -80,6 +80,12 @@ protected async Task<LSP.Location[]> GetDefinitionAsync(LSP.TextDocumentPosition
             // local functions
             static bool ShouldInclude(INavigableItem item, bool typeOnly)
             {
+                if (item.Glyph is Glyph.Namespace)
```

Comment:
nit: Should include this in the switch statement below

Also as a side q: Is there just another task to better refactor the gotodef services in Roslyn or do you expect that the LSP handler will forever have this type of "apply or don't apply" logic in it?

Reply:
this needs to go before the switch - the switch is only hit when typeOnly = true.

I considered having this check in the IFindDefinitionService.  But I think really for a namespace the service is returning the right thing (the locations of the namespace definition), So making the service return nothing might not be appropriate (we just don't want to display them).  But I could be convinced otherwise

Reply:
Ah didn't catch the typeOnly check 👍.

> I considered having this check in the IFindDefinitionService. But I think really for a namespace the service is returning the right thing (the locations of the namespace definition), So making the service return nothing might not be appropriate (we just don't want to display them). But I could be convinced otherwise

I think I was more curious about if there's an underlying provider model or set of options for go-to-definition capabilities that would enable you to configure which "providers" or options are applicable for LSP. A lot like what you [do for completion](https://github.com/dotnet/roslyn/blob/6753a64a2ca34fab9916f7ad034b895d906a1f7e/src/Features/LanguageServer/Protocol/Handler/Completion/CompletionHandler.cs#L239-L242).